### PR TITLE
Using PHP7 strict_types everywhere in the source

### PR DESCRIPTION
**Pull request by evert**

### CODE REVIEW

File: lib/CalDAV/SharingPlugin.php [hunk 1]
```diff
@@ -304,6 +304,7 @@ function httpPost(RequestInterface $request, ResponseInterface $response) {
 
                 if ($url) {
                     $writer = $this->server->xml->getWriter();
+                    $writer->contextUri = $request->getUrl();
```
Comment (staabm) [inline lib/CalDAV/SharingPlugin.php]: Unrelated change?

Comment (evert) [inline lib/CalDAV/SharingPlugin.php]: No, this is initialized as `null` and caused some issues further down the line.
